### DISCUSSION TITLE
two-factor hardware keys no longer working in firefox

### DISCUSSION THREAD

**Discussion by afhpayne** [community/community#23037] [New to GitHub]
<p>Hi, does anyone know did something change with u2f at Github?  As of today, Firefox 60.5.2esr is being blocked from using Yubikeys to login.</p>

**Comment by afhpayne** [comment, marked as answer]
<p>Resolved by github today.</p>

**Comment by mflis** [comment]
<p>Seeing this image right now.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/beec7e537195a05b8107f52bcc26744decf0d4d6.png" data-download-href="/uploads/short-url/reZp45bQzlPmT95GCLMG1AtHAW2.png?dl=1" title="Screenshot from 2019-03-19 20-42-21.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/15031715/181074290-f6420c05-e8b1-4b1a-94ef-19a4271bf251.png" alt="Screenshot from 2019-03-19 20-42-21.png" data-base62-sha1="reZp45bQzlPmT95GCLMG1AtHAW2" width="408" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/beec7e537195a05b8107f52bcc26744decf0d4d6_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot from 2019-03-19 20-42-21.png</span><span class="informations">412×504 16.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Background info from Firefox Troubleshooting Information:</p>
<p>Version 65.0.2<br>
Build ID 2019022821563</p>
<p>OS Linux 5.0.2-arch1-1-ARCH</p>

**Comment by Sancus** [comment]
<p>This is not resolved.</p>
<p>Right now if I get a login prompt from GitHub while I’m using Firefox, it asks me to enter my password. If I change my user agent to Chrome, still using Firefox, it properly adds that I can press a button on my security key to log in.</p>
<p>So U2F is still being blocked in Firefox based purely on user agent.</p>

**Comment by qwrtln** [comment]
<p>Looks like everything except for Chrome is blocked. I wonder what could be the reason for this.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Sorry for the trouble you’ve all had here. We’ve recently pushed some changes that should have re-enabled the work-around you were using previously. So please do try signing in again as you had done prior to these last few days.</p>
<p>If you continue to encounter issues though, please do let us know!</p>

**Comment by mgarstecki** [comment]
<p>I still can’t register my Yubikey through Firefox, there is this message on the security page : "This browser doesn’t support the FIDO U2F standard yet. We recommend updating to the latest <a href="https://www.google.com/chrome/browser/desktop/" rel="nofollow noopener">Google Chrome</a> to start using security key devices. ".</p>
<p>I succesfully registered and use my Yubikey as U2F through this browser on other providers (AWS, …).</p>

**Comment by petreeftime** [comment]
<p>I cannot register any keys with Firefox 66.0.3. <code>u2f</code> is enabled in <code>about:config</code> and it definitely works on other sites. Github seems to only support Chrome for registering keys for some reason.</p>

**Comment by heisian** [comment]
<blockquote>
<p><a class="mention" href="/andreagriffiths11">@andreagriffiths11</a> wrote:</p>
</blockquote>
<blockquote>
<p>Sorry for the trouble you’ve all had here. We’ve recently pushed some changes that should have re-enabled the work-around you were using previously. So please do try signing in again as you had done prior to these last few days.</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>If you continue to encounter issues though, please do let us know!</p>
</blockquote>
<p>I am still seeing an issue:</p>
<p>macOS 10.13.6</p>
<p>Firefox 66.0.3 (64-bit)</p>
<p>security.webauth.u2f === true</p>
<p>This browser doesn’t support the FIDO U2F standard yet. We recommend updating to the latest <a href="https://www.google.com/chrome/browser/desktop/" rel="nofollow noopener">Google Chrome</a> to start using security key devices.</p>

**Comment by an** [comment]
<p>Still not working, Firefox 66.0.5 64-bit Win10, see screenshot, the notice is displayed when on Security page. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/35537eb9820d696aa1aee8a1e1d5ec9ade9122e2.png" data-download-href="/uploads/short-url/7BKbYBqep4JwpMoY95DsPxfVBmO.png?dl=1" title="firefoxYubikey.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/102708/181074297-c930da10-bc6f-490b-bc30-db1e5012c2b0.png" alt="firefoxYubikey.png" data-base62-sha1="7BKbYBqep4JwpMoY95DsPxfVBmO" width="690" height="386" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/35537eb9820d696aa1aee8a1e1d5ec9ade9122e2_2_690x386.png, https://aws1.discourse-cdn.com/github/original/2X/3/35537eb9820d696aa1aee8a1e1d5ec9ade9122e2.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/3/35537eb9820d696aa1aee8a1e1d5ec9ade9122e2.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/35537eb9820d696aa1aee8a1e1d5ec9ade9122e2_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">firefoxYubikey.png</span><span class="informations">990×555 54 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by tdurk93** [comment]
<p>I’m on Firefox 67 and also seeing that GitHub is telling me to switch to Chrome. I checked that u2f (and WebAuthn) is enabled in the browser’s about:config. Once I switched the user-agent to Chrome, I was able to successfully register my hardware token. Can you unblock Firefox (at least the latest versions) from being able to register/use tokens?</p>

**Comment by regnarock** [comment]
<p>Couldn’t connect using Firefox either!</p>
<p>I also checked security.webauth.u2f is at true.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/9c4d45f76b830eb4b013004c117a600a15c626be.png" data-download-href="/uploads/short-url/miI2E5ey5bdEndndFxRR17WDsJw.png?dl=1" title="image.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/4267223/181074302-9391e6d5-1334-414b-aa74-8b6ef83cadcb.png" alt="image.png" data-base62-sha1="miI2E5ey5bdEndndFxRR17WDsJw" width="690" height="458" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/9/9c4d45f76b830eb4b013004c117a600a15c626be_2_690x458.png, https://aws1.discourse-cdn.com/github/optimized/2X/9/9c4d45f76b830eb4b013004c117a600a15c626be_2_1035x687.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/9/9c4d45f76b830eb4b013004c117a600a15c626be_2_1380x916.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/9c4d45f76b830eb4b013004c117a600a15c626be_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image.png</span><span class="informations">1666×1106 340 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by keeakita** [comment]
<p>As others have reported, this is not actually resolved. Firefox 67 just released as stable, which has u2f enabled as a default. I’m still seeing the message even after successfully using u2f on other sites:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/3dd304a9c63e4c79db765ba326dfe9abd97438ac.png" data-download-href="/uploads/short-url/8OVegEB5f2qK2pjwr4Bd4CtPBla.png?dl=1" title="Screenshot_20190528_163303.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/1066487/181074307-ca83247c-3007-4778-87cb-72ae229e15a7.png" alt="Screenshot_20190528_163303.png" data-base62-sha1="8OVegEB5f2qK2pjwr4Bd4CtPBla" width="690" height="157" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/3dd304a9c63e4c79db765ba326dfe9abd97438ac_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot_20190528_163303.png</span><span class="informations">734×168 14.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>More info:</p>
<ul>
<li>
<p>Firefox 67.0 (64-bit) fedora - 1.0</p>
</li>
<li>
<p>Fedora 29 x64</p>
</li>
</ul>

**Comment by NicoPy** [comment]
<p>Tried to register a U2F key today. Same problem as others.</p>
<p>Using Firefox 67 on Winfows 10.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Thanks for reaching out, and apologies for the delayed response here, we wanted to make sure there was not any updates on this topic.</p>
<p>Can I ask if the <code>security.webauth.u2f</code> flag has been enabled in Firefox? You can find it by entering <code>about:config</code> into the browser’s address bar. From there, you can search for <code>security.webauth.u2f</code> and make sure it’s set to <code>true</code>. Once that has been set, you should be able to add a new security key.</p>
<p>If you have any issues with that however, it may be worth setting up in Google Chrome first to configure a new key, then you would be able to sign-in with Firefox.</p>
<p>Sorry, I didn’t have a more definitive answer to provide here but we’re limited in our ability to troubleshoot issues on your browser, especially while official support is still not something we’re quite able to offer at this time. </p>

**Comment by rtyer** [comment]
<p>This still is an issue. Can we get an ETA for resolution?</p>

**Comment by greduan** [comment]
<p>Can confirm that logging in works with Firefox, but I first have to setup the key in Chrome.</p>

**Comment by flocke** [comment]
<p>I can confirm as well that it doesn’t work with Firefox 67.0.4 on Ubuntu 19.04 as well as on Arch Linux.</p>

**Comment by holstvoogd** [comment]
Stil broken.. wtf @github-staff

**Comment by Frank-Krick** [comment]
still broken

**Comment by Frank-Krick** [comment]
or rather again?
